.github/templates/linux_ci_workflow.yml.in

```diff
@@ -0,0 +1,147 @@
+          SCCACHE_MAX_JOBS=$(( $(nproc) - 1 ))
+          MEMORY_LIMIT_MAX_JOBS=8  # the "large" resource class on CircleCI has 32 CPU cores, if we use all of them we'll OOM
+          MAX_JOBS=$(( ${SCCACHE_MAX_JOBS} > ${MEMORY_LIMIT_MAX_JOBS} ? ${MEMORY_LIMIT_MAX_JOBS} : ${SCCACHE_MAX_JOBS} ))
+          docker run \
+            -e BUILD_ENVIRONMENT \
+            -e MAX_JOBS \
+            -e SCCACHE_BUCKET \
+            -e SKIP_SCCACHE_INITIALIZATION=1 \
+            -e TORCH_CUDA_ARCH_LIST \
+            --security-opt seccomp=unconfined \
+            --cap-add=SYS_PTRACE \
+            --tty \
+            --user jenkins \
+            -v ${GITHUB_WORKSPACE}:/var/lib/jenkins/workspace \
+            -w /var/lib/jenkins/workspace \
+            ${DOCKER_IMAGE} \
+            sh -c 'sudo chown -R jenkins . && .jenkins/pytorch/build.sh'
+      - name: Chown workspace
+        run: |
+          # Ensure the working directory gets chowned back to the current user
+          docker run --rm -v "$(pwd)":/v -w /v alpine chown -R "$(id -u):$(id -g)" .
+      - uses: actions/upload-artifact@v2
+        name: Store PyTorch Build Artifacts
+        with:
+          name: ${{ env.BUILD_ENVIRONMENT }}
+          retention-days: 30
+          if-no-files-found: error
+          path:
+            ${{ github.workspace }}/dist/**/*
+            ${{ github.workspace }}/build/**/*
+      - name: Clean up docker images
+        if: always()
+        run: |
+          # Prune all of the docker images
+          docker system prune -af
+  test:
+    runs-on: linux.2xlarge
+    needs:
+      - calculate-docker-image
+      - build
+    env:
+      DOCKER_IMAGE: ${{ needs.calculate-docker-image.outputs.docker_image }}
+    steps:
+      - name: Chown workspace
+        run: |
+          # Ensure the working directory gets chowned back to the current user
+          docker run --rm -v "$(pwd)":/v -w /v alpine chown -R "$(id -u):$(id -g)" .
+      - name: Checkout PyTorch
+        uses: actions/checkout@v2
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+          submodules: recursive
+      - name: Log in to ECR
+        run: |
+          $(aws ecr get-login --no-include-email --region us-east-1)
+      - name: Pull docker image
+        run: |
+          docker pull "${DOCKER_IMAGE}"
+      - uses: actions/download-artifact@v2
+        name: Download PyTorch Build Artifacts
+        with:
+          name: ${{ env.BUILD_ENVIRONMENT }}
+      - name: Test PyTorch
+        run: |
+          docker run \
+            -e BUILD_ENVIRONMENT \
+            -e IN_CI \
+            --security-opt seccomp=unconfined \
+            --cap-add=SYS_PTRACE \
+            --tty \
+            --user jenkins \
+            -v ${GITHUB_WORKSPACE}:/var/lib/jenkins/workspace \
+            -w /var/lib/jenkins/workspace \
+            ${DOCKER_IMAGE} \
+            sh -c 'sudo chown -R jenkins . && pip install dist/*.whl && .jenkins/pytorch/test.sh'
```

Comment:
All this chowning back and forth is such a pain. You should just update the docker images to have a user at the correct user ID that GitHub actions expects and then use that instead of jenkins

Reply:
We could, but then it'd require us to build a new image which is something I don't really want to tackle with this PR. Perhaps in a follow up we can create a separate stage with a separate EC2 user

Reply:
As well we also do this in our current circleci configuration